library/core-components/components/auto-layout/auto-layout.styles.ts

```diff
@@ -49,31 +49,40 @@ export const setPosition = (
   return out;
 };
 
-export type StyleProps = Pick<
-  React.ComponentProps<typeof RadiusAutoBox>,
+/** When no item spacing is specified, figma defaults to 10px */
+const FIGMA_DEFAULT_SPACING = '10px';
+
+/**
+ * The component props that are used to generate the styles. The props passed as
+ * the third generic argument are required (eg. when we know a prop is assigned
+ * a default value in the component).
+ */
+// ? Is this too confusing? Would it be better to split the types out to make it more clear?
+export type StyleProps = PickWithRequired<
+  React.ComponentProps<typeof RadiusAutoLayout>,
   | 'direction'
   | 'space'
-  | 'clippedContent'
-  | 'alignment'
-  | 'width'
-  | 'height'
   | 'padding'
   | 'opacity'
   | 'fill'
   | 'stroke'
   | 'strokeWidth'
   | 'strokeAlign'
   | 'cornerRadius'
-  | 'isParent'
-  | 'absolutePosition'
   | 'x'
   | 'y'
-  | 'horizontalConstraint'
-  | 'verticalConstraint'
   | 'dropShadow'
   | 'innerShadow'
   | 'layerBlur'
-  | 'backgroundBlur'
+  | 'backgroundBlur',
+  | 'alignment'
+  | 'width'
+  | 'height'
+  | 'clippedContent'
+  | 'isParent'
+  | 'absolutePosition'
+  | 'horizontalConstraint'
+  | 'verticalConstraint'
 >;
```

Comment:
Is this confusing? Would it be better to split the logic up even if it is more verbose?